.github/workflows/development.yml

```diff
@@ -0,0 +1,147 @@
+#
+#
+# Following Azure services are consumed:
+# 1. Azure resource group - https://learn.microsoft.com/en-us/cli/azure/group?view=azure-cli-latest#az-group-create
+# 2. Azure container registry - https://learn.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-create
+# 3. Azure container app environment - https://learn.microsoft.com/en-us/azure/container-apps/environment
+# 4. Azure container app - https://learn.microsoft.com/en-us/azure/container-apps/containers
+# 
+#
+# Execution
+# *********
+# GHA is only invoked when following conditions are satisfied:
+# 1. Push to the `dev` branch only.
+# 2. Any modifications inside `src` directory
+
+name: Deployment 🚀
+run-name: APIM dev deployment for ${{ github.repository }}
+
+on:
+  push:
+    branches: [ APIM-149-ia-c-dev-environment ]
+    paths:
+    - 'src/**'
+
+env:
+  product: apim
+  environment: dev
+  timezone: 'Europe/London'
+  # Base artifact
+  from: latest
+
+jobs:
+
+# 1. Setup deployment variables
+  setup:
+    name: Setup 🔧
+    runs-on: [ self-hosted, APIM, deployment ]
+    outputs:
+      product: ${{ env.product }}
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
+    steps:
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. MDM micro-service deployment
+  mdm:
+    name: MDM 📦️
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    env:
+      name: mdm
+      environmet: ${{ needs.setup.outputs.environment }}
+      cr: cr${{ needs.setup.outputs.product }}${{ needs.setup.outputs.environment }}${{ vars.REGION }}001
+    runs-on: [ self-hosted, APIM, deployment ]
+    steps:
+
+    - name: Repository 🗃️
+      uses: actions/checkout@v3
+
+    - name: Azure 🔐
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: CLI 📝
+      run: echo acr=$(az acr show -n ${{ env.cr }} --query loginServer | tr -d '"') >> $GITHUB_ENV
+
+    - name: Defaults ✨
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          # Basic
+          az configure --defaults location=${{ vars.REGION }}
+          az configure --defaults group=rg-${{ env.product }}-${{ env.environment }}-001
+
+    - name: ACR 🔐
+      uses: azure/docker-login@v1
+      with:
+        login-server: ${{ env.acr }}
+        username: ${{ secrets.ACR_USERNAME }}
+        password: ${{ secrets.ACR_PASSWORD }}
+    
+    - name: Artifacts 🗃️
+      run: |
+        docker build . \
+        -t ${{ env.acr }}/${{ env.name }}:${{ github.sha }} \
+        -t ${{ env.acr }}/${{ env.name }}:${{ env.from }}
+        docker push ${{ env.acr }}/${{ env.name }}:${{ github.sha }}
+        docker push ${{ env.acr }}/${{ env.name }}:${{ env.from }}
+
+    - name: Revisions 🔀
```

Comment:
Just realised now, are we not able to use the official GH Action for deployment?
> https://github.com/marketplace/actions/azure-container-apps-build-and-deploy

Reply:
Let's address this as part of a different pr